### PR TITLE
Allow reshaping an inhomogeneous system to a single unit cell

### PR DESCRIPTION
**Pull request by kbarros**
This feature will support the upcoming LSWT-KPM capability developed by @hlane33.

### CODE REVIEW

File: src/Reshaping.jl [hunk 1]
```diff
@@ -32,19 +34,35 @@ function reshape_supercell(sys::System{N}, shape) where N
 end
 
 
-# Transfer homogeneous interactions from `sys.origin` to reshaped `sys`
-function set_interactions_from_origin!(sys::System{N}) where N
-    origin = sys.origin
-    new_ints  = interactions_homog(sys)
-    orig_ints = interactions_homog(origin)
+# Transfer homogeneous interactions from `orig` to reshaped `sys`.
```
Comment (ddahlbom) [inline src/Reshaping.jl:37]: Should "homogenous" be removed from the description here?

File: src/Reshaping.jl [hunk 2]
```diff
@@ -8,6 +8,8 @@ periodicity of a requested supercell. The columns of the ``3×3`` integer matrix
 original crystal lattice vectors.
 """
 function reshape_supercell(sys::System{N}, shape) where N
+    is_homogeneous(sys) || error("Cannot reshape system with inhomogeneous interactions.")
```
Comment (ddahlbom) [inline src/Reshaping.jl:11]: The point of this is to prevent users from doing this? In other words, you have to know about `reshape_supercell_aux`.

**Review by ddahlbom — COMMENTED** [review]
This looks like a nice solution to me.

File: src/Reshaping.jl (hunk 2 above)
Comment (kbarros) [reply src/Reshaping.jl:11]: Yes. I assume we don't want people reshaping inhomogeneous systems in general. This PR makes an exception for LSWT, which is a very specialized use case.

**Comment by kbarros** [comment]
I am thinking this PR is still correct... probably we need to iterate with a real test case.

File: src/Reshaping.jl (hunk 1 above)
Comment (kbarros) [reply src/Reshaping.jl]: Updated

**Comment by kbarros** [comment]
Had a helpful conversation with @ddahlbom. We realized there are some possible simplifications. Now `reshape_supercell_aux` should work more generally. Ready for testing by @hlane33 !

File: src/Reshaping.jl (hunk 2 above)
Comment (kbarros) [reply src/Reshaping.jl:11]: Update -- there's a bit more flexibility now. For example, with the latest commits, it becomes possible to call `repeat_periodically` on an inhomogeneous system.

**Comment by kbarros** [comment]
Merging now to facilitate @hlane33's work. It needs more testing, but the use cases are somewhat specialized, so we can deal with any errors as they arise.

**Pull request merged**